src/foam/nanos/fs/File.js

```diff
@@ -69,9 +70,64 @@ foam.CLASS({
         }
         return url;
       }
+    },
+    {
+      class: 'Blob',
+      name: 'data',
+      javaGetter:`
+        if ( dataIsSet_ ) return data_;
+        if ( ! SafetyUtil.isEmpty(this.getDataString()) ) {
+          String encodedString = this.getDataString().split(",")[1];
+          BlobService blobStore = new foam.blob.BlobStore();
+          byte[] decodedBytes = Base64.getDecoder().decode(encodedString);
+          InputStream is = new ByteArrayInputStream(decodedBytes);
+          InputStreamBlob blob = new foam.blob.InputStreamBlob(is, decodedBytes.length);
+          return blob;
+        } else {
+          return null;
+        }
+      `,
+      getter: function(){
+        if ( typeof this.dataString != 'undefined' && this.dataString != '' ) {
```

Comment:
You can do foam.Undefined.isInstance(this.dataString). That way you don't need to know five different methods in JS to tell if something is the correct type because FOAM has the equivalent method for each type.